### PR TITLE
Updating the maven plugin for a better usability

### PR DESCRIPTION
**Pull request by ehsavoie**
Allowing the Maven plugin to use a folder with classes and to ouput the transformed classes toa folder so they can be used to produce a new artefact

### CODE REVIEW

File: api/src/main/java/org/wildfly/extras/transformer/ArchiveTransformer.java [hunk 1]
```diff
@@ -125,35 +122,18 @@ protected ResourceTransformer newResourceTransformer() throws IOException {
...
 
-    private static void writeBytes(final OutputStream os, final byte[] clazz, final boolean closeStream) throws IOException {
-        try {
-            os.write(clazz);
-        } finally {
-            if (closeStream) {
-                safeClose(os);
-            }
-        }
+    public boolean supportClassTransformation() {
```
Comment (scottmarlow) [inline api/src/main/java/org/wildfly/extras/transformer/ArchiveTransformer.java:136]: Is this supposed to return false file transformers that do not support transforming standalone class files?  Should it install be called `supportClassFileTransformation` or `canTransformIndividualClassFile`?

Comment (ehsavoie) [reply api/src/main/java/org/wildfly/extras/transformer/ArchiveTransformer.java:136]: yes you are correct :) I'll rename it

**Comment by scottmarlow** [comment]
I was able to use Batavia built with this pull request + Galleon to build WildFly and run the Jakarta EE 9 Platform TCK signature tests and didn't see any issues as a result of these changes.

File: impl/ecl/src/main/java/org/wildfly/extras/transformer/eclipse/ArchiveTransformerImpl.java [hunk 2]
```diff
@@ -124,4 +124,9 @@ private boolean transform(String[] args, boolean silent) throws IOException {
         return false;
     }
 
+    @Override
+    public boolean canTransformIndividualClassFile() {
```
Comment (scottmarlow) [inline impl/ecl/src/main/java/org/wildfly/extras/transformer/eclipse/ArchiveTransformerImpl.java:128]: I thought this is what is meant but wasn't 100% sure before.  Thanks for changing the method name to be more clear.  I believe that the no-dependency + ASM transformers could also handle this as well but we haven't been actively working on those transformers (we could in the future).

**Review by scottmarlow — APPROVED** [review]

**Review by ropalka — APPROVED** [review]

**Comment by scottmarlow** [comment]
Thanks @ropalka!

**Pull request merged**